api/docs/release.dox

```diff
@@ -173,6 +173,10 @@ Further non-compatibility-affecting changes include:
  - Added opnd_create_immed_double(), opnd_get_immed_double() and
    opnd_is_immed_double() to enable the creation and handling of double
    precision floating-point operands.
+ - Added -reset_at_created_thread_count option to allow triggering reset when
```

Comment:
Still confused: I am not seeing why we have to add a whole new option that is so similar to the existing -reset_at_nth_thread.  The use case here is test apps that we control so we can make the thread count exactly what we want.  I'm also not really understanding the difference: this seems like a duplicate option.

Reply:
PR #4491 has a comment thread on this. Basically, `-reset_at_nth_thread n` triggers reset when the **active** thread count reaches `n`. Because of #4496, we cannot use n=2. We also cannot use n>2 because `linux.thread-reset` has at most two threads active at any given time (to keep synchronisation simple for deterministic print ordering).

So, we need `-reset_at_created_thread_count n` which triggers reset when the **created** thread count reaches `n`. Note that #4496 affects this option with n=2 too. But the good thing about this option is that we can use n=3 to reproduce the reset issue.

Reply:
Hmm.. I can probably put a workaround in suite/tests/linux/thread.c. Let me try.

Reply:
This doesn't seem like the best solution: adding a whole new duplicate option just to work around an assert we want to fix anyway and to avoid a 3rd thread in a tiny test app.

Reply:
I modified the reset tests to have 3 concurrent threads so that we can use `-reset_at_nth_thread 3`. But the tests seem flaky now: `linux.thread-reset` fails 3/100 times, `linux.clone-reset` fails 29/100 times. 

This is surprising because previously, with `-reset_at_created_thread_count 3`, I didn't see any flakiness in 100 runs (all pass). But maybe the same race was present then too.

Reply:
So before, both the tests passed 100/100 times. Not sure what's causing the flakiness now (maybe something to do with the fact that we have more concurrent threads running now than before). 

I think it's better to just fix the missing pieces in #4496 first. Then if the flakiness is still relevant, I can get back to it